libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -179,6 +181,65 @@ bool IRGeneratorForStatements::visit(FunctionCall const& _functionCall)
 	return false;
 }
 
+bool IRGeneratorForStatements::visit(InlineAssembly const& _inlineAsm)
+{
+	auto const& loc = _inlineAsm.location();
+
+	solAssert(loc.start != 1 && loc.end != -1, "Expected source location");
+
+	vector<string> params, retParams, args;
+	// Prepare parameters and arguments
+	for (auto const& pair: _inlineAsm.annotation().externalReferences)
+	{
+		auto varDecl = dynamic_cast<VariableDeclaration const*>(pair.second.declaration);
+		solUnimplementedAssert(!!varDecl, "");
+
+		args.emplace_back(m_context.variableName(*varDecl));
+		params.emplace_back(varDecl->name());
+		retParams.emplace_back("ret_" + varDecl->name());
+	}
+
+	// Skip the "assembly" literal
+	int blockStart = loc.start + string("assembly").size();
+
+	stringstream funcName, assignVars;
+
+	funcName <<	"inline_assembly_" << loc.start << "_" << loc.end;
```

Comment:
Maybe use the AST node ID as suffix instead? Also I'm wondering about whether this can result in name collisions and how we could avoid that... what if you happen to define a function called ``inline_assembly_*`` in the assembly block? What if you call e.g. an ``abi_decode_*`` function in it (which you shouldn't be able to)? All that is probably pretty complicated in the end, but I think it might make sense to have a preliminary version available now that just ignores all that stuff...

Reply:
Maybe to solve all that naming business we need to use a customized ``AsmPrinter`` that adjusts identifiers instead of just copying the original source string in the end after all :-).